### PR TITLE
ADJob Podtemplates for tigera-intrusion-detecion with corresponding updated ClusterRoles

### PR DESCRIPTION
**Pull request by freecaykes**
## Description
New resources that were added:
- an ADJob template for training 
- an ADJob template for detection
- update to the ClusterRole for IntrusionDetection's Service Account to allow fetching and editting Jobs / CronJobs, Podtemplate
  

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: https://github.com/tigera/operator/pull/1568/commits/ec512f6fe5850f8248bf827fece7559b037f9a17

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: https://github.com/tigera/operator/pull/1568/commits/ec512f6fe5850f8248bf827fece7559b037f9a17

File: pkg/render/logstorage.go
Comment (freecaykes) [reply pkg/render/logstorage.go]: trickled the `ElasticSearchPort` up to render/logstorage level since it was defined lower in render/logstorage/elastic pkg before

File: config/enterprise_versions.yml
Comment (Brian-McM) [inline config/enterprise_versions.yml:61]: This needs to be `anomaly_detection_jobs` to match what's in the process repo (which builds our hash releases).

File: pkg/crds/calico/crd.projectcalico.org_globalnetworkpolicies.yaml
Comment (freecaykes) [reply pkg/crds/calico/crd.projectcalico.org_globalnetworkpolicies.yaml]: removed with the merge from master

File: pkg/render/intrusion_detection.go
Comment (Brian-McM) [inline pkg/render/intrusion_detection.go:330]: Don't you need create / watch / update as well?

File: pkg/render/logstorage/esgateway/esgateway.go
Comment (Brian-McM) [inline pkg/render/logstorage/esgateway/esgateway.go:49]: Could you please put this file back the way it was and just leave it as duplicated? We're trying to avoid importing the render package into these subpackages, and duplication is acceptable to avoid it.

We're ultimately trying to put all the components in the render package in their own subpackages to clean things up.

**Review by Brian-McM — CHANGES_REQUESTED** [review]

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go:330]: we are just using an instantaneous Get for now in IDS controller for now since we will only read values  and create jobs / cronjobs out of them

Comment (Brian-McM) [reply pkg/render/intrusion_detection.go:330]: Oh right, these are the permissions for the IDS controller not the operator.

**Review by Brian-McM — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
